### DISCUSSION TITLE
Error: more than one Image found for name ubuntu-20.04

### DISCUSSION THREAD

**Discussion by TimHeckel** [mysticaltech/terraform-hcloud-kube-hetzner#773] [Q&A]
Hi All! I'm running a version of this project from last October, so perhaps that's the real problem here - but I'm trying to update my nodepool agent count with the below command and I'm getting the below error

terraform apply -auto-approve

```
with module.kube-hetzner.module.agents["0-4-sb-agent"].hcloud_server.server,
│   on ../modules/host/main.tf line 22, in resource "hcloud_server" "server":
│   22: resource "hcloud_server" "server" {
```

Not sure if this is exactly where the images are being sourced from, but I did curl the hetzner API and found only one image for `ubuntu-20.04` so I'm not sure what's up:

```
url \
        -H "Authorization: Bearer xxx" \
        'https://api.hetzner.cloud/v1/images'
```

Any pointers would be super helpful. Thanks!

**Comment by TimHeckel** [comment]
Ah I see the issue is due to two different architectures for the ubuntu images when I run `hcloud image list` -- the problem is that I don't know how  to specify the `arch` option (or whatever) in my `main.tf` file (which is now legacy compared to the more recent versions of the kube-hetzner project):

```
resource "hcloud_server" "server" {
  name               = local.name
  image              = "ubuntu-20.04"
  arch                 = "x86"  # THIS DOES NOT WORK - not sure how to specify the image architecutre...
  rescue             = "linux64"
  server_type     = var.server_type
  location           = var.location
  ssh_keys         = var.ssh_keys
  firewall_ids     = var.firewall_ids
  placement_group_id = var.placement_group_id
  user_data        = data.cloudinit_config.config.rendered
  ...
}
```

**Comment by TimHeckel** [comment]
Ah, it appears [the issue stems](https://www.reddit.com/r/hetzner/comments/12jrh42/terraform_with_hetzner/) from the  the underlying [terraform-provider-hcloud](https://github.com/hetznercloud/terraform-provider-hcloud) repo was pinned to version `1.35.2` in my lock file, so I was able to re-init terraform after changing this constraint to `1.38.1` and this fixed this issue, but unfortunately it brought another one as the process gets further: 

```
Error: remote-exec provisioner error
│
│   with module.kube-hetzner.null_resource.agents["0-4-sb-agent"],
│   on ../agents.tf line 65, in resource "null_resource" "agents":
│   65:   provisioner "remote-exec" {
│
│ error executing "/tmp/terraform_178438531.sh": Process exited with status 1
```

After a little more digging, the issue isn't in running the above script, it seems to be this: 
```
module.kube-hetzner.null_resource.agents["0-4-sb-agent"] (remote-exec): Ok: return value of 0.
module.kube-hetzner.null_resource.agents["0-4-sb-agent"] (remote-exec): Committing changes:
module.kube-hetzner.null_resource.agents["0-4-sb-agent"] (remote-exec): Conflicting name type transition rules
module.kube-hetzner.null_resource.agents["0-4-sb-agent"] (remote-exec): Binary policy creation failed at /var/lib/selinux/targeted/tmp/modules/400/k3s/cil:135
module.kube-hetzner.null_resource.agents["0-4-sb-agent"] (remote-exec): Failed to generate binary
module.kube-hetzner.null_resource.agents["0-4-sb-agent"]: Still creating... [10s elapsed]
module.kube-hetzner.null_resource.agents["0-4-sb-agent"] (remote-exec): /sbin/semodule:  Failed!
```

I have confirmed that this ubuntu instance is indeed x86, so I don't know what else may be causing this binary failure...

**Comment by mysticaltech** [comment]
@TimHeckel Too much has happenned since October, just upgrade, we have an Upgrade section in the readme, and most upgrade issues have already dealt within, search for them on the project when you are faced with an error, and if blocked, just tag me I will try to help.

**Reply by mysticaltech** [reply]
Then, after upgrading, you will be able to scale up the nodepool count without issues without issues.

**Comment by TimHeckel** [comment]
Hi @mysticaltech  - thanks, that's what I did, and it works great! Thanks for working on this amazing project.

**Reply by mysticaltech** [reply]
Thanks @TimHeckel, happy to hear that! 🙏
